elisp/ghc-info.el

```diff
@@ -31,10 +31,25 @@
 ;;; type
 ;;;
 
+(defface ghc-type-region
+  '((default :inherit region)
+    (((background light))
+     :background "LightBlue")
+    (((background dark))
+     :background "RoyalBlue"))
+  "Face used to mark region selected by `ghc-show-type'.")
+
 (defvar ghc-type-overlay nil)
 
 (make-variable-buffer-local 'ghc-type-overlay)
 
+(defvar ghc-type-map
```

Comment:
If you're overriding all these operations why not just set the region temporarily, that'll eliminate the type highlighting vs. region confusion. It looks like the region highlighting doesn't survive a `C-c C-t` invocation but `C-x C-x C-x C-x` will restore it. So if we restore the original region plus highlighting when the user moves point that should be even better.